File: Array/Sliding Window/Fixed/java/Leetcode643.java
class Leetcode643 {
    static {
        for (int i = 0; i < 100; i++) {
            findMaxAverage(new int[1], 1);
        }
    }
    public static double findMaxAverage(int[] nums, int k) {
        
        int n = nums.length; 
        int sum = 0;
        
        for (int i = 0; i < k; i++) {
            sum += nums[i];
        }
        
        int maxSum = sum;
        
      
        for (int i = k; i < n; i++) {
            sum += nums[i] - nums[i - k];
            if (sum > maxSum) {
                maxSum = sum; 
            }
        }
        
        return (double) maxSum / k;
    }
    public static void main(String[] args) {
        int[] nums = new int[]{1,12,-5,-6,50,3};
        System.out.println(findMaxAverage(nums, 4));
    }
}

File: Array/Sliding Window/Fixed/python/Leetcode643.py
class Solution:
    def findMaxAverage(self, nums: list[int], k: int) -> float:
        n = len(nums)
        if k <= 0 or k > n:
            return 0.0

        window_sum = sum(nums[0:k])
        max_avg = window_sum / k

        for i in range(k, n):
            window_sum += nums[i] - nums[i - k]
            max_avg = max(max_avg, window_sum / k)

        return float(max_avg)
        

def main():
    nums = [1,12,-5,-6,50,3]
    obj = Solution()
    print(obj.findMaxAverage(nums, 4))

if __name__ == "__main__":
    main()